client/app/services/session.service.js

```diff
@@ -117,6 +124,26 @@
       return angular.isDefined(serviceFeature);
     }
 
+    function isAnyActionAllowed(actions, productFeatures) {
+      for (var i = 0; i <= actions.length; i++) {
+        if (angular.isDefined(actions[i])) {
```

Comment:
@h-kataria This method does not seem to work due to the reasons below -
1. `actions[i]` is merely a string here and `angular.isDefined` would always return a `true` for that
2. What `angular.isDefined` needs to look at is the object `productFeatures.atomic_catalogitem_edit`

So this is what I recommend -
1. Let `actions` be an array of the feature names
```javascript
var actions = ["atomic_catalogitem_edit",
               "atomic_catalogitem_new",
               "catalog_items_view",
               "catalogitem_edit",
               "catalogitem_new"];
```
2. Make sure you reference these feature names via the `productFeatures` object
```javascript
if (angular.isDefined(productFeatures[actions[i]]))
```